src/inference/src/dev/icompiled_model.cpp

```diff
@@ -50,6 +50,21 @@ ov::ICompiledModel::ICompiledModel(const std::shared_ptr<const ov::Model>& model
 
         std::unordered_map<std::shared_ptr<ov::descriptor::Tensor>, std::shared_ptr<ov::descriptor::Tensor>> tensor_map;
         for (const auto& param : model->get_parameters()) {
+            {
+                const auto& p_shape = param->get_partial_shape();
+                const auto& layout = param->get_layout();
+                if (p_shape.rank().is_static() && p_shape.rank() != 0 && ov::layout::has_batch(layout)) {
```

Comment:
That was my initial intention, but there is a bunch of tests that rely on this possibility to have some dimensions set to zero (e.g. _EyeTransformationTestsP.eye_decompose_). I'm not sure how correct is this, but the behavior in this case seems to be valid and thus I guess it could be used somewhere else except the tests, so I would not cut this functionality.